libraries/Microsoft.Bot.Builder/OAuth/IConnectorClientProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+
+namespace Microsoft.Bot.Builder.OAuth
+{
+    /// <summary>
+    /// Abstraction to build connector clients.
+    /// </summary>
+    public interface IConnectorClientProvider
```

Comment:
If we refactor the adapter and auth layers in R10, should this interface be here or in the Connector dll?

Reply:
Decided offline that @carlosscastro going to rename the interface and keep it in this DLL as it's for concealing a higher-level implementation.

Additionally, the ServiceUrl, ClaimsIdentity and Audience are not all the requisite pieces for making a ConnectorClient, an IServiceClientCredentialProvider is required as well.

This interface will likely not be ported to JS and Python as createConnectorClient is a public method in those languages.

Reply:
Since nothing is really "private" or "protected" in Python, it should be a minimal concern.  i.e., If it had been notated as private, people can still access and override it.  It's more of a pirate guideline.  Meant to be broken if you feel like it.  Like types.

Reply:
With that said, not sure I see a problem anyway.  It appears that now this method is declared in an interface, and calls the internal private version.  But it has the same signature.  So really aren't we just declaring it in an abstract class, and overriding it with the current impl (for Python)?

Reply:
Yes, forget about the explicit implementatin (which is C# sugar). The goal of this is that the OAuthPrompt deals with an IConnectorClientBuilder rather than an adapter, which gives us the possibility of decoupling from the adapter. Particularly considering that we want to do serious changes in the adapter and auth layer :)

Reply:
IMHO, between this, PR microsoft/botbuilder-dotnet#3818 and the large amount of code in the BotFrameworkAdapter, there is definitely a decent amount of work for comparable value in decoupling.

Reply:
We want to get to a point where all those 1000 lines of auth prompt code are completely outside the adapter. We'll get there.